apps/code-examples/src/app/code-examples/popovers/popover/basic/popover-demo.component.spec.ts

```diff
@@ -0,0 +1,111 @@
+import { HarnessLoader } from '@angular/cdk/testing';
+import { TestbedHarnessEnvironment } from '@angular/cdk/testing/testbed';
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+import { NoopAnimationsModule } from '@angular/platform-browser/animations';
+import { expect } from '@skyux-sdk/testing';
+import { SkyPopoverAlignment, SkyPopoverPlacement } from '@skyux/popovers';
+import { SkyPopoverHarness } from '@skyux/popovers/testing';
+
+import { PopoverDemoComponent } from './popover-demo.component';
+import { PopoverDemoModule } from './popover-demo.module';
+
+describe('Basic popover', () => {
+  async function setupTest(options?: {
+    openPopover: boolean;
+    titleText?: string;
+    alignment?: SkyPopoverAlignment;
+    placement?: SkyPopoverPlacement;
+    dismissOnBlur?: boolean;
+  }): Promise<{
+    popoverHarness: SkyPopoverHarness | null;
+    fixture: ComponentFixture<PopoverDemoComponent>;
+    loader: HarnessLoader;
+  }> {
+    const fixture = TestBed.createComponent(PopoverDemoComponent);
+    const loader = TestbedHarnessEnvironment.documentRootLoader(fixture);
+    let open = true;
+
+    if (options) {
+      open = options.openPopover;
+      fixture.componentInstance.popoverAlignment = options.alignment;
+      fixture.componentInstance.popoverPlacement = options.placement;
+      fixture.componentInstance.popoverTitle = options.titleText;
+      fixture.componentInstance.dismissOnBlur = options.dismissOnBlur;
+    }
+
+    fixture.detectChanges();
+    await fixture.whenStable();
+
+    if (open) {
+      const button = document.querySelector<HTMLButtonElement>('.sky-btn');
```

Comment:
I'm thinking we're going to need a harness for the `skyPopover` directive. That way, you could query for a specific trigger button using `data-sky-id`, and then have `open()` and `isOpen()` methods the consumer can use in their tests.

Reply:
Refactored